lib/cache/cache.go

```diff
@@ -229,6 +229,7 @@ func ForNode(cfg Config) Config {
 		{Kind: types.KindClusterAuthPreference},
 		{Kind: types.KindSessionRecordingConfig},
 		{Kind: types.KindRole},
+		{Kind: types.KindNode},
```

Comment:
@rosstimothy @fspmarshall @espadolini This is going to cause problems with performance and/or network utilization in large clusters, right? A node that runs EC2 auto-discovery needs to query other nodes in the cluster to understand which nodes have already joined.

@lxea Can we make it so this is included in watch configuration only if the node runs auto-discovery? Otherwise if each node will watch other nodes is going to destroy large clusters.